core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -168,22 +169,28 @@ private void updateHttpRequestInInterceptorContext(SdkHttpFullRequest request, E
     }
 
     /**
-     * We do not sign if the Auth SchemeId is smithy.api#noAuth.
+     * We sign if a signer is provided is not null.
      *
      * @return True if request should be signed, false if not.
      */
-    private boolean shouldSign(SelectedAuthScheme<?> selectedAuthScheme) {
-        // TODO: Should this string be a constant somewhere. Similar logic is used in AuthSchemeInterceptors.
-        return !"smithy.api#noAuth".equals(selectedAuthScheme.authSchemeOption().schemeId());
+    private boolean shouldSign(Signer signer) {
+        return signer != null;
     }
 
     /**
-     * We sign if a signer is provided is not null.
-     *
-     * @return True if request should be signed, false if not.
+     * Returns true if we should use the selected out scheme attribute for signing. Returns true if the client has overridden
+     * the signer using the advanced configuration setting or if the no selected auth scheme.
      */
-    private boolean shouldSign(Signer signer) {
-        return signer != null;
+    private boolean shouldUseSelectedAuthScheme(RequestExecutionContext context) {
+        return !hasClientConfiguredSigner() &&
+               context.executionAttributes().getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME) != null;
+    }
+
+    /**
+     * Returns true if the client configuration has an override for the signer.
+     */
+    private boolean hasClientConfiguredSigner() {
+        return dependencies.clientConfiguration().option(SdkAdvancedClientOption.SIGNER) != null;
```

Comment:
😬 You're right, I guess I just tunneled on fixing the failures instead of looking closer. I will remove these and leave at least the following test failing;

* [`ProfileFileConfigurationTest.profileIsHonoredForCredentialsAndRegion`](https://github.com/aws/aws-sdk-java-v2/blob/a3341396ddd52df4171246d2add662fec264ef3c/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/ProfileFileConfigurationTest.java#L44)
* [`AsyncSignerOverrideTest.test_signerOverriddenForStreamingInput_takesPrecedence`](https://github.com/aws/aws-sdk-java-v2/blob/a3341396ddd52df4171246d2add662fec264ef3c/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/AsyncSignerOverrideTest.java#L45)
* [`AsyncOperationCancelTest.testEventStreamingOperation`](https://github.com/aws/aws-sdk-java-v2/blob/03f2db714bf987b59e8e5b7f6ec2d9b46f3126dc/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/protocolrestjson/AsyncOperationCancelTest.java#L96) (this one is commented out since otherwise the test just hangs).

Reply:
These 3 tests are re-enabled in https://github.com/aws/aws-sdk-java-v2/pull/4264.